src/lib/encoder.ts

```diff
@@ -80,6 +81,55 @@ const decodeDataSourceWithHash = (value: string): URLDataWithHash => {
   return { hashFunction: hashFunction.name, hash: dataHash, url: dataSource };
 };
 
+const encodeCompactBytesArray = (values: string[]): string => {
+  let compactBytesArray = '0x';
+
+  for (let i = 0; i < values.length; i++) {
+    if (values[i] === '')
+      throw new Error(
+        `Couldn't encode, value at index ${i} is '', expected '0x'`,
+      );
+
+    if (!isHex(values[i]))
+      throw new Error(`Couldn't encode, value at index ${i} is not hex`);
+
+    if (values[i].length > 65_535 * 2 + 2)
+      throw new Error(
+        `Couldn't encode bytes[CompactBytesArray], value at index ${i} exceeds 65_535 bytes`,
+      );
+
+    const numberOfBytes = (values[i].length - 2) / 2;
+    const hexNumber = padLeft(numberToHex(numberOfBytes), 4);
+
+    compactBytesArray += stripHexPrefix(hexNumber) + stripHexPrefix(values[i]);
+  }
+
+  return compactBytesArray;
+};
+
+const decodeCompactBytesArray = (compactBytesArray: string): string[] => {
+  if (!isHex(compactBytesArray))
+    throw new Error("Couldn't decode, value is not hex");
+
+  let pointer = 2;
```

Comment:
I think this might also fail by using non `0x` prefixed hex strings as it will pass the `isHex` and then you will remove 2 chars